oe_contact_forms.module

```diff
@@ -537,3 +539,23 @@ function oe_contact_forms_mail_alter(&$message) {
   // Change subject for corporate mail, copy and reply.
   $message['subject'] = $contact_form->getThirdPartySetting('oe_contact_forms', 'email_subject');
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_access().
+ */
+function oe_contact_forms_contact_form_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if ($operation !== 'view') {
+    return AccessResult::neutral();
+  }
+
+  // Leave default contact forms untouched.
+  if (!$entity->getThirdPartySetting('oe_contact_forms', 'is_corporate_form', FALSE)) {
+    return AccessResult::neutral();
+  }
+
+  if ($account->hasPermission('access corporate contact form')) {
+    return AccessResult::allowed();
```

Comment:
We need to add cache by permissions here

Reply:
done